### DISCUSSION TITLE
gd-plug can install itself

### DISCUSSION THREAD

**Discussion by lihop** [imjp94/gd-plug#3] [General]
Just something cool I noticed.

If you install gd-plug via the AssetLib, you can then add gd-plug to the plug.gd like so:
```
extends "res://addons/gd-plug/plug.gd"

func _plugging():
    plug("imjp94/gd-plug")
```
Then reinstall using the `force` flag to overwrite the AssetLib files:
```
godot --no-window -s plug.gd install force
```
Now gd-plug is installed and managed by itself!

I think that's pretty neat.

**Comment by imjp94** [comment]
Cool, I don't even know it is possible!

But, I recommend to upgrade gd-plug itself through `upgrade` command, `godot --no-window -s plug.gd upgrade`
As this setup would limit user to always install with `force` command, which I suggest to avoid unless you know what you are doing as it might overwrite your project files.

For example, if someone force install and accidentally left out `install_root` when including all files from plugin repository, gd-plug will overwrite all project files with identical names and directory like project.godot, icon.png, etc.
```gdscript
plug("HungryProton/scatter", {"include": ["."]}) # install_root missing
```
While by default, gd-plug will terminate the installation and avoid any loss of project files.

But making gd-plug manage itself seems to be a nice feature to consider, maybe it should check for update everytime executing `status` command
